project/frontend/src/components/Authentication.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+
+const AuthContext = React.createContext({});
+
+function Authentication(props) {
+  const [GoogleAuth, setGoogleAuth] = useState(undefined);
+  const [currentUser, setCurrentUser] = useState(undefined);
+  const [signedIn, setSignedIn] = useState(false);
+
+  const context = {
+    GoogleAuth: { get: GoogleAuth, set: setGoogleAuth },
+    currentUser: { get: currentUser, set: setCurrentUser },
+    signedIn: { get: signedIn, set: setSignedIn },
```

Comment:
Instead of creating a context object here, can we just put this in `React.createContext({})` above?

Reply:
I don't think we can because this context object needs to be recreated every time the `Authentication` component re-renders due to state changes.

Also the parameter passed to `React.createContext(...)` is apparently only given to components listening to that context that don't have a `Provider` above them in the tree. I've put the `Authentication` really high in the tree so I don't think there's any way a consumer of `AuthContext` wouldn't have `Authentication` above it.